cmd/nerdctl/run.go

```diff
@@ -965,6 +965,11 @@ func parseKVStringsMapFromLogOpt(cmd *cobra.Command, logDriver string) (map[stri
 			delete(logOptMap, logging.MaxFile)
 		}
 	}
+	if value, ok := logging.LogOptValidateFuncs[logDriver]; ok {
+		if err := value(logOptMap); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Not really a Fun of putting funcs in Map. I think we can expose this logic by a switch case
I would recommend sthg like this (just a pseudo code) :

```suggestion
switch logDriver {
	case "json-file" :
		LogOptValidateFunc = logging.JSONFileLogOptsValidate
	case "journald":
	        LogOptValidateFunc = logging.JournalLogOptsValidate
}

logging.LogOptValidateFunc(logOptMap)

```